storage-proofs/post/src/fallback/vanilla.rs

```diff
@@ -352,6 +386,10 @@ impl<'a, Tree: 'a + MerkleTreeTrait> ProofScheme<'a> for FallbackPoSt<'a, Tree>
             partition_proofs.push(Proof { sectors: proofs });
         }
 
+        ensure!(faulty_sectors.is_empty(), {
```

Comment:
As this isn't really an *early* return, I'd find it easier to understand if it would be written as:

```rust
if (faulty_sectors.is_empty()) {
    Ok(partition_proofs)
} else {
    Err(Error::FaultySectors(faulty_sectors.into_iter().collect()))
}
```

But that's a matter of taste, hence no need to change it.

Reply:
I actually agree. I changed it because what you wrote (and I tried first) wouldn't compile. I figured out how to fix it, though, so thanks for the nudge.

```
        if faulty_sectors.is_empty() {
            Err(Error::FaultySectors(faulty_sectors.into_iter().collect()).into())
        } else {
            Ok(partition_proofs)
        }
```